lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -410,16 +412,21 @@ public function processParameterValue($value)
             return $value;
         }
 
-        if (is_object($value) && $this->_em->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value))) {
+        if ($value instanceof Mapping\ClassMetadata) {
+            return $value->name;
+        }
+
+        if ( ! is_object($value)) {
+            return $value;
```

Comment:
It seems like this path is not being covered by tests, can you please add one?

Reply:
I know, it looks much better now. However, in the previous algorithm this scenario was implicitly covered and now it's not.

![image](https://user-images.githubusercontent.com/201963/48423073-352fc580-e760-11e8-90ea-14d70461b6eb.png)

Reply:
@alcaeus yes and that's exactly why I'm asking for one extra unit test to consider this PR ready to go. We're doing a lot of refactoring for v3 and it would be a shame if the this scenario doesn't work anymore...

Reply:
Final test pushed.